src/sentry/taskworker/client/client.py

```diff
@@ -167,22 +167,23 @@ def __init__(
         self._temporary_unavailable_host_timeout = temporary_unavailable_host_timeout
 
         self._health_check_settings = health_check_settings
-        self._requests_since_touch_lock = threading.Lock()
-        self._requests_since_touch = (
-            0 if health_check_settings is None else health_check_settings.requests_per_touch
-        )
+        self._timestamp_since_touch_lock = threading.Lock()
+        self._timestamp_since_touch = 0.0
 
     def _emit_health_check(self) -> None:
         if self._health_check_settings is None:
             return
 
-        with self._requests_since_touch_lock:
-            self._requests_since_touch += 1
-            if self._requests_since_touch < self._health_check_settings.requests_per_touch:
+        with self._timestamp_since_touch_lock:
```

Comment:
You may want to acquire the lock only when you are about to touch the file. This should reduce lock contention.